# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.cpp
# Commit: 62480de389ff
# Full Hash: 62480de389ffba353998de24760993b2e1531232
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth. CLOSED TREE
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r f2e486f1be17 -r 62480de389ff js/src/wasm/WasmTypes.cpp
--- a/js/src/wasm/WasmTypes.cpp	Thu Aug 27 10:30:46 2020 +0000
+++ b/js/src/wasm/WasmTypes.cpp	Thu Aug 27 10:53:04 2020 +0000
@@ -634,10 +634,11 @@
 
 /* static */
 DebugFrame* DebugFrame::from(Frame* fp) {
-  MOZ_ASSERT(fp->instance()->code().metadata().debugEnabled);
+  MOZ_ASSERT(
+      GetNearestEffectiveTls(fp)->instance->code().metadata().debugEnabled);
   auto* df =
       reinterpret_cast<DebugFrame*>((uint8_t*)fp - DebugFrame::offsetOfFrame());
-  MOZ_ASSERT(fp->instance() == df->instance());
+  MOZ_ASSERT(GetNearestEffectiveTls(fp)->instance == df->instance());
   return df;
 }
 
@@ -658,6 +659,10 @@
 #endif
 }
 
+Instance* DebugFrame::instance() const {
+  return GetNearestEffectiveTls(&frame_)->instance;
+}
+
 GlobalObject* DebugFrame::global() const {
   return &instance()->object()->global();
 }